pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -629,13 +673,72 @@ func (b *OFBridge) RetryInterval() time.Duration {
 	return b.retryInterval
 }
 
+func (b *OFBridge) queryTableFeatures() {
+	mpartRequest := &openflow13.MultipartRequest{
+		Header: openflow13.NewOfp13Header(),
+		Type:   openflow13.MultipartType_TableFeatures,
+		Flags:  0,
+	}
+	mpartRequest.Header.Type = openflow13.Type_MultiPartRequest
+	mpartRequest.Header.Length = mpartRequest.Len()
+	tableFeatureChan := make(chan *openflow13.MultipartReply)
+	b.mpartRplChs[mpartRequest.Header.Xid] = tableFeatureChan
+	go b.processTableFeatures(mpartRequest.Header.Xid, tableFeatureChan)
+	b.ofSwitch.Send(mpartRequest)
+}
+
+func (b *OFBridge) processTableFeatures(xid uint32, ch chan *openflow13.MultipartReply) {
+	header := openflow13.NewOfp13Header()
+	header.Type = openflow13.Type_MultiPartRequest
+	requests := make([]*openflow13.MultipartRequest, 0)
+	tableCount := 0
+	for tableCount < int(LastTableID-1) {
+		select {
+		case rpl := <-ch:
+			request := &openflow13.MultipartRequest{
+				Header: header,
+				Type:   openflow13.MultipartType_TableFeatures,
+				Flags:  1,
+			}
+			for _, body := range rpl.Body {
```

Comment:
what does this mean? Each reply may contain one or more tables?
Can you add some comments for better understanding?

Reply:
Correct, we first query the existing table features on OVS, and only changes the name property on the table which is used in our pipeline. Each MultipartReply mesasge might include one or many TableFeature messages, while each TableFeature is focusing on one table.